### PR TITLE
Bump lingo to 0.3 and remove hack associated with #333.

### PR DESCRIPTION
**Pull request by patrickt**
- [x] Depends upon a release of lingo-0.3.0.0.

Now that lingo's language detection returns all possible results for a
given filename, we're able to abolish the workaround where we parsed
the string `"Hack"` into PHP. We have to check ourselves for spurious
results associated with Hack, TSX, and Markdown files, but that's
fitting and proper, rather than doing it behind the scenes in lingo.

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -33,6 +33,11 @@ package semantic-source
 package semantic-tags
   ghc-options: -Werror
 
+source-repository-package
+  type: git
+  location: https://github.com/tclem/lingo-haskell
+  tag: 59b19fd7251bd247245dbb67859507391764270f
```
Comment (patrickt) [inline cabal.project]: Gonna nuke this when lingo-0.3.0.0 lands.

File: src/Data/Language.hs [hunk 2]
```diff
@@ -93,7 +93,15 @@ extensionsForLanguage language = T.unpack <$> maybe mempty Lingo.languageExtensi
 
 -- | Return a language based on a FilePath's extension.
 languageForFilePath :: FilePath -> Language
-languageForFilePath path = maybe Unknown (textToLanguage . Lingo.languageName) (Lingo.languageForPath path)
+languageForFilePath path =
+  let spurious lang = lang `elem` [ "Hack" -- .php files
+                                  , "GCC Machine Description" -- .md files
+                                  , "XML" -- .tsx files
```
Comment (patrickt) [inline src/Data/Language.hs]: Not sure why lingo thinks .tsx files could be XML, but a workaround was straightforward (and I added this to the tests).

Comment (robrix) [reply src/Data/Language.hs]: [This is why](https://github.com/tclem/lingo-haskell/blob/master/languages.yml#L5593).

Comment (patrickt) [reply src/Data/Language.hs]: wow wtf

Comment (robrix) [reply src/Data/Language.hs]: something something XML and force something

File: semantic.cabal [hunk 3]
```diff
@@ -71,7 +71,7 @@ common dependencies
                      , text ^>= 1.2.3.1
                      , these >= 0.7 && <1
                      , unix ^>= 2.7.2.2
-                     , lingo >= 0.2.0.0
+                     , lingo >= 0.3.0.0
```
Comment (robrix) [inline semantic.cabal]: This is unbounded at the top end; let’s use `^>=`.

**Pull request merged**